tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/move_instruction.h

```diff
@@ -54,16 +55,17 @@ class MoveInstruction
   using Ptr = std::shared_ptr<MoveInstruction>;
   using ConstPtr = std::shared_ptr<const MoveInstruction>;
 
-  MoveInstruction(Waypoint waypoint, MoveInstructionType type, const std::string& profile = "DEFAULT");
+  MoveInstruction(Waypoint waypoint,
+                  MoveInstructionType type,
+                  const std::string& profile = "DEFAULT",
+                  ManipulatorInfo manipulator_info = ManipulatorInfo());
 
   void setWaypoint(Waypoint waypoint);
   const Waypoint& getWaypoint() const;
 
-  void setTCP(const Eigen::Isometry3d& tcp);
-  const Eigen::Isometry3d& getTCP() const;
-
-  void setWorkingFrame(std::string working_frame);
-  const std::string& getWorkingFrame() const;
+  void setManipulatorInfo(ManipulatorInfo info);
+  const ManipulatorInfo& getManipulatorInfoConst() const;
```

Comment:
You can remove the `Const` from the name.

Reply:
When they are the same name it errors saying I am trying to overload based on return type.

Reply:
done